File: SingleNumber/SingleNumber.java
class Solution {
    public int singleNumber(int[] nums) {
        Set s = new HashSet<Integer>();
        for (int i = 0; i < nums.length; i++) {
            if (s.contains(nums[i])) {
                s.remove(nums[i]);
            } else {
                s.add(nums[i]);
            }
        }
        return (int)(s.iterator().next());
    }
}

File: SingleNumber/SingleNumber.py
import collections

class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        #Bit Manipulation (one liner!)
        #return reduce(lambda x, y: x ^ y, nums)
        
        #Hash Table with dict comprehension xd
        return {v: k for k, v in collections.Counter(nums).items()}[1]
        
        
